src/schematools/types.py

```diff
@@ -527,14 +498,31 @@ def use_dimension_fields(self) -> bool:
         return self._parent_schema.use_dimension_fields
 
     @property
-    def temporal(self) -> Optional[Dict[str, Union[str, Dict[str, List[str]]]]]:
-        """Return the temporal info from the dataset schema """
-        return self._parent_schema.fetch_temporal(field_modifier=lambda x: x)
+    def temporal(
+        self, field_modifier: Optional[Callable] = None
+    ) -> Optional[Dict[str, Union[str, Dict[str, List[str]]]]]:
```

Comment:
This was a pre-existing return type. I might have even introduced it. It was probably the best i could do at the time, not understanding any of the temporal logic. I still don't. But it looks horrendous and doesn't not help much in conveying what is actually returned here.

You seem to have a better understanding of the temporal logic, and are touching another dependent project (DSO-API). This is a good time to introduce a better data structure here instead of this amorphous dict.

What I am saying is that this seems a perfect moment to introduce a `dataclass` representing the temporal data in schema's. You should:
* define that `dataclass`, 
* provide proper docstrings, 
* and use it here to return it.

That should put us on the path to a better structured and typed Amsterdam Schema representating in Python.

@jjmurre What do you think?

Reply:
+1 for a dataclass. The datastructure that needs to be represented is:
    
    temporal: {
      identifier: "volgnummer",
      dimensions: {
        geldigOp: [
          "beginGeldigheid",
          "eindGeldigheid"
       ]
    }
   }

So it would be quite straighforward to make this a dataclass.

Reply:
Or actually two dataclasses I think, one separate dataclass for the inner `dimensions`.

Reply:
Definitely add a `TemporalSchema` class here, in line with the other classes (hence I wouldn't use a dataclass but schema dict like the rest). This also simplifies the DSO-API code. Instead of `.temporal.get('identifier', None)` you can just use `.temporal.identifier` as that's the actual field provided by this class.

Reply:
Would a dict of tuples be better for the dimensions? Or can a dimension have a different shape?

Reply:
I would say temporality is always defined with a `start` and and `end`. So a 2-item list or a 2-item tuple. I agree with Diederik that, to keep things consistent, use a schemadict subclass.

Reply:
Bedoel je deze: `SchemaType(UserDict)`? Maar temporality is geen schema. En heeft dus geen id en type properties.